src/js/components/Home/Countries.js

```diff
@@ -2,39 +2,63 @@ import React, {Component} from "react";
 import Map from "./Map";
 import './MapStyles.sass'
 import ReactFlagsSelect from 'react-flags-select';
+import Geocode from "react-geocode";
 
+// function onSelectFlag(countryCode){             //TODO: Function to be implemented
+//     console.log(countryCode)
+// }
+// set Google Maps Geocoding API for purposes of quota management. Its optional but recommended.
+Geocode.setApiKey("AIzaSyBiCz9njYP8nYxEUEQLzltCM8kAnm7z6To");
 
 class Mapbuttons extends Component {
     constructor() {
         super()
         this.state = {center: [0, 0]}
+        this.getCountryCode = this.getCountryCode.bind(this);
     }
-    changeCenter = center => () => {
-        this.setState({center})
-    }
+    // changeCenter(center) {
+    //     this.setState({center})
+    // }
 
-     onSelectFlag=countryCode=>() =>{
-        console.log(countryCode);
-        return this.changeCenter([-122.4194, 37.7749])
+    getCountryCode(country) {
+        console.log(country);
+        Geocode.fromAddress(country).then(
+            response => {
+                const { lat, lng } = response.results[0].geometry.location;
+                 this.setState({center: [lat,lng] });
+                 console.log(lat);
+                 console.log(lng);
+                 console.log(this.state.center[0]);
+                // this.changeCenter([lat, lng])
+            },
+            error => {
+                console.error(error);
+            }
+        );
     }
 
+    // onSelectFlag(countryCode){
+    //     console.log(countryCode)
+    // }
 
     render() {
         return (
             <div className="container-fluid">
                 <div className="row">
-                    <div className="col-9">
+                    <div className="col-md-9 col-12 ">
                         <Map center={this.state.center}/>
                     </div>
-                    <div className="col-3">
+                    {/*<button onClick={()=>this.test("South Africa")}>press</button>*/}
```

Comment:
gc

Reply:
I'll remove all useless gc comments, thank you for the feedback.